src/test/java/org/gridsuite/directory/server/DirectoryTest.java

```diff
@@ -1212,15 +1205,33 @@ private void checkElementNameExistInDirectory(UUID parentDirectoryUuid, String e
                         .andExpect(status().is(expectedStatus.value()));
     }
 
-    private void deleteElement(UUID elementUuidToBeDeleted, UUID elementUuidHeader, String userId, boolean isRoot, boolean isPrivate) throws Exception {
+    private void deleteElement(UUID elementUuidToBeDeleted, UUID elementUuidHeader, String userId, boolean isRoot, boolean isPrivate, boolean isStudy, int numberOfChildStudies) throws Exception {
         mockMvc.perform(delete("/v1/elements/" + elementUuidToBeDeleted)
                 .header("userId", userId))
                         .andExpect(status().isOk());
 
+        Message<byte[]> message;
+        MessageHeaders headers;
+        // assert that the broker message has been sent a delete study
+        if (isStudy) {
+            message = output.receive(1000);
+            assertEquals("", new String(message.getPayload()));
+            headers = message.getHeaders();
+            assertEquals(userId, headers.get(HEADER_USER_ID));
+            assertEquals(UPDATE_TYPE_STUDY_DELETE, headers.get(HEADER_UPDATE_TYPE));
+            assertEquals(elementUuidToBeDeleted, headers.get(HEADER_STUDY_UUID));
+        } else {
+            //empty the queue of all delete study notif
+            for (int i = 0; i < numberOfChildStudies; i++) {
+                message = output.receive(1000);
+                headers = message.getHeaders();
+                assertEquals(UPDATE_TYPE_STUDY_DELETE, headers.get(HEADER_UPDATE_TYPE));
```

Comment:
Maybe check also that HEADER_USER_ID header is userId ?

Reply:
sure why not